pkg/cephfs/volume.go

```diff
@@ -121,26 +141,17 @@ func purgeVolume(volID volumeID, adminCr *credentials, volOptions *volumeOptions
 		return fmt.Errorf("error mounting ceph root: %v", err)
 	}
 
-	defer unmountAndRemove(cephRoot)
-
-	if err := os.Rename(volRoot, volRootDeleting); err != nil {
-		return fmt.Errorf("couldn't mark volume %s for deletion: %v", volID, err)
-	}
-
-	if err := os.RemoveAll(volRootDeleting); err != nil {
-		return fmt.Errorf("failed to delete volume %s: %v", volID, err)
-	}
-
 	return nil
 }
 
-func unmountAndRemove(mountPoint string) {
-	var err error
-	if err = unmountVolume(mountPoint); err != nil {
-		klog.Errorf("failed to unmount %s with error %s", mountPoint, err)
+func unmountCephRoot(volID volumeID) {
+	cephRoot := getCephRootPathLocal(volID)
+
+	if err := unmountVolume(cephRoot); err != nil {
+		klog.Errorf("failed to unmount %s with error %s", cephRoot, err)
```

Comment:
question: if unmount fails do we need to remove the directory?
if we continue to do `os.Remove()` won't we end up in stale mount points?

Reply:
If unmount fails, removing /should/ fail too. Good point though, no need to delete it if unmounting failed - will fix.